add nekoray and cpr

GoldsrcSource opened the pull request.

GoldsrcSource commented on net-proxy/nekoray/nekoray-4.2.12.ebuild:
```diff
@@ -0,0 +1,33 @@
+	doexe "${BUILD_DIR}"/nekoray
+	newbin "${FILESDIR}"/"${PN}".sh "${PN}"
+	dolib.so "${BUILD_DIR}"/libqhotkey.so.0.1
+	dolib.so "${BUILD_DIR}"/libqhotkey.so.0
```
possible conflict with other packages. can this be done better?

peeweep replied:
先安装到 /usr/$(get_libdir)/${PN}/ 解决和 dev-libs/qhotkey 的 file conflicts ？

GoldsrcSource replied:
patchelf --replace-needed libqhotkey.so.0 ./libqhotkey.so.0.1 "${ED}"/usr/lib/"${PN}"/nekoray || die
will this work?

GoldsrcSource replied:
it actually worked..

peeweep commented on net-proxy/nekoray/nekoray-4.2.12.ebuild:
```diff
@@ -0,0 +1,33 @@
+DEPEND="dev-cpp/abseil-cpp
+		dev-libs/protobuf:=
+		dev-qt/qtbase:6[widgets,network]
+		dev-cpp/yaml-cpp
```
漏了 cpr ?

peeweep commented on net-proxy/nekoray/nekoray-4.2.12.ebuild:
```diff
@@ -0,0 +1,36 @@
+
+src_install() {
+	exeinto "/usr/lib/${PN}/"
+	doexe "${BUILD_DIR}"/nekoray
```
BUILD_DIR 应该是不需要的。

GoldsrcSource replied:
```
>>> Install net-proxy/nekoray-4.2.12 into /var/tmp/portage/net-proxy/nekoray-4.2.12/image
!!! doexe: nekoray does not exist
 * ERROR: net-proxy/nekoray-4.2.12::gentoo-zh failed (install phase):
 *   doexe failed
 * 
 * If you need support, post the output of `emerge --info '=net-proxy/nekoray-4.2.12::gentoo-zh'`,
 * the complete build log and the output of `emerge -pqv '=net-proxy/nekoray-4.2.12::gentoo-zh'`.
 * The complete build log is located at '/var/tmp/portage/net-proxy/nekoray-4.2.12/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/net-proxy/nekoray-4.2.12/temp/environment'.
 * Working directory: '/var/tmp/portage/net-proxy/nekoray-4.2.12/work/nekoray-4.2.12'
 * S: '/var/tmp/portage/net-proxy/nekoray-4.2.12/work/nekoray-4.2.12'
 * QA Notice: file does not exist:
 * 
 *      doexe: nekoray does not exist
```

peeweep commented on net-proxy/nekoray/nekoray-4.2.12.ebuild:
```diff
@@ -0,0 +1,36 @@
+	patchelf --replace-needed libqhotkey.so.0 /usr/lib/"${PN}"/libqhotkey.so.0.1 "${ED}"/usr/lib/"${PN}"/nekoray || die
+	newbin "${FILESDIR}"/"${PN}".sh "${PN}"
+	into "/usr/lib/${PN}/"
+	doexe "${BUILD_DIR}"/libqhotkey.so.0.1
```
同上，BUILD_DIR 应该是不需要的。

GoldsrcSource replied:
```
>>> Install net-proxy/nekoray-4.2.12 into /var/tmp/portage/net-proxy/nekoray-4.2.12/image
!!! doexe: nekoray does not exist
 * ERROR: net-proxy/nekoray-4.2.12::gentoo-zh failed (install phase):
 *   doexe failed
 * 
 * If you need support, post the output of `emerge --info '=net-proxy/nekoray-4.2.12::gentoo-zh'`,
 * the complete build log and the output of `emerge -pqv '=net-proxy/nekoray-4.2.12::gentoo-zh'`.
 * The complete build log is located at '/var/tmp/portage/net-proxy/nekoray-4.2.12/temp/build.log'.
 * The ebuild environment file is located at '/var/tmp/portage/net-proxy/nekoray-4.2.12/temp/environment'.
 * Working directory: '/var/tmp/portage/net-proxy/nekoray-4.2.12/work/nekoray-4.2.12'
 * S: '/var/tmp/portage/net-proxy/nekoray-4.2.12/work/nekoray-4.2.12'
 * QA Notice: file does not exist:
 * 
 *      doexe: nekoray does not exist
```

GoldsrcSource committed:
net-proxy/nekoray: new package, add 4.2.12

Signed-off-by: G-Src <gsrc@fsfans.club>

net-proxy/nekobox-core: new package, add 4.2.12

Signed-off-by: G-Src <gsrc@fsfans.club>

dev-libs/cpr: new package, add 1.11.2

Signed-off-by: G-Src <gsrc@fsfans.club>
